src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -111,7 +118,11 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   async componentDidMount() {
-    const { store, dependentStores, isClusterScoped } = this.props;
+    const { store, dependentStores, isClusterScoped, isConfigurable, tableId, renderTableHeader } = this.props;
+    if (isConfigurable && tableId) {
+      this.columnProps = new Set(userStore.preferences.tablesConfiguration[tableId]
+        ?? renderTableHeader.map(cellProps => { return cellProps.className; }));
```

Comment:
With the above this is simpler since an empty set implies "show all columns".
```suggestion
      this.hiddenColumnNames = new Set(userStore.preferences?.tablesConfiguration?.[tableId]);
```